uriparser 1.0.2 build error with glibc <2.29

On our x86_64 Linux/chromebrew systems we are seeing this build error:

```
UriMemory.c:109:12: error: returning ‘int’ from a function with return type ‘void *’ makes pointer from integer without a cast [-Wint-conversion]
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,7 +52,6 @@ set(URIPARSER_SO_REVISION   2)
 set(URIPARSER_SO_AGE        2)
 
 include(CheckCCompilerFlag)
-include(CheckCSourceCompiles)
 include(CheckFunctionExists)
 include(CheckLibraryExists)
 include(CheckSymbolExists)
@@ -136,43 +135,6 @@ set(URIPARSER_EXTRA_COMPILE_FLAGS)
 check_c_compiler_flag("-fvisibility=hidden" URIPARSER_COMPILER_SUPPORTS_VISIBILITY)
 if(URIPARSER_COMPILER_SUPPORTS_VISIBILITY)
     set(URIPARSER_EXTRA_COMPILE_FLAGS "${URIPARSER_EXTRA_COMPILE_FLAGS} -fvisibility=hidden")
-endif()
-
-#
-# Environment checks
-#
-check_c_source_compiles("
-    #include <features.h>
-
-    #if ! defined(__GLIBC__)
-    # error libc is not glibc
-    #endif
-
-    int main(void) {
-        return 0;
-    }"
-    _URIPARSER_GLIBC_USED)
-
-# Reject glibc <2.29 as unsupported
-if(_URIPARSER_GLIBC_USED)
-    check_c_source_compiles("
-        #include <features.h>
-
-        #if ! __GLIBC_PREREQ(2, 29)
-        # error glibc too old to be supported
-        #endif
-
-        int main(void) {
-            return 0;
-        }"
-        _URIPARSER_GLIBC_SUPPORTED)
-
-    if(NOT _URIPARSER_GLIBC_SUPPORTED)
-        message(SEND_ERROR
-            "Use with glibc <2.29 is unsupported."
-            " "
-            "Please compile against a different C runtime library.")
-    endif()
 endif()
 
 #

--- a/src/UriMemory.c
+++ b/src/UriMemory.c
@@ -50,6 +50,11 @@
 #    define _DEFAULT_SOURCE 1
 #  endif
 
+// For glibc <2.29
+#  if !defined(_GNU_SOURCE)
+#    define _GNU_SOURCE 1
+#  endif
+
 // For NetBSD (stdlib.h revision 1.122 of 2020-05-26)
 #  if defined(__NetBSD__) && !defined(_OPENBSD_SOURCE)
 #    define _OPENBSD_SOURCE 1

